### PR TITLE
backport the histogram & json report file feature

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/12738/

**Comment by dothebart** [comment]
tests blue.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12779/

**Comment by dothebart** [comment]
tests blue except for currently unavailable macs.

File: arangosh/Benchmark/BenchFeature.cpp [hunk 1]
```diff
@@ -451,31 +554,40 @@ bool BenchFeature::writeJunitReport(BenchRunResult const& result) {
...
+  builder.add("requestTime", VPackValue(result.requestTime));
   std::cout << "Request/response duration (per thread): " << std::fixed
             << (result.requestTime / (double)_concurrency) << " s" << std::endl;
+  builder.add("requestResponseDurationPerThread", VPackValue(result.requestTime / (double)_concurrency));
 
   std::cout << "Time needed per operation: " << std::fixed
-            << (result.time / _realOperations) << " s" << std::endl;
-
+            << (result.time / _operations) << " s" << std::endl;
+  builder.add("timeNeededPerOperation", VPackValue(result.time / _realOperations));
```
Comment (mpoeter) [inline arangosh/Benchmark/BenchFeature.cpp:567]: The line above uses `result.time / _operations`, but here we add `result.time / _realOperations`.

Comment (dothebart) [reply arangosh/Benchmark/BenchFeature.cpp:567]: good spot!

**Comment by dothebart** [comment]
tests blue except not working mac compile.

**Review by mpoeter — APPROVED** [review]

**Comment by dothebart** [comment]
superseeded by https://github.com/arangodb/arangodb/pull/13133

**Pull request closed without merging**